### ISSUE TITLE
error 500

### ISSUE THREAD

**Issue by badsmoke** [j6s/mailcow-exporter#1]
hello i get this error, i have the newest mailcow version installed

empty json array '[]'
```
020/09/05 18:41:41 https://mail.domain.tld/api/v1/get/mailq/all
2020/09/05 18:41:41 &errors.errorString{s:"Error while updating metrics of provider.Mailq: \"Could not parse JSON: \\\"json: cannot unmarshal object into Go value of type []provider.queueResponseItem\\\"\""}
```

**Comment by j6s** [comment]
That's interesting. Can you try issuing a manual API request in order to verify, that the returned data is in fact JSON?

Something like this should do:

```bash
$ curl 'https://mail.domain.tld/api/v1/get/mailq/all' -H 'X-Api-Key: YOUR-API-KEY-HERE'
```

Things to look for in the response body
* Is it JSON?
* Is it a JSON array with items of the following format:

```json
  {
    "queue_name": "deferred",
    "queue_id": "0E348C0104",
    "arrival_time": 1599158539,
    "message_size": 22076,
    "sender": "foo@mail.domain.tld",
    "recipients": [
      "..."
    ]
  }
```

**Comment by j6s** [comment]
Looking at possible error sources, I have discovered, that currently authentication errors are not handled correctly: They return the following JSON

```json
{
  "type": "error",
  "msg": "authentication failed"
}
```

which in fact would lead to a JSON marshalling error similar to the one you posted. Not sure if this is the same problem or not though. Please report back with your findings.

**Comment by j6s** [comment]
I also just pushed a commit to `main` (should be in docker hub soon) which makes errors more verbose in order to help track issues with API requests down.

**Comment by badsmoke** [comment]
the result from the curl command is a empty json struct 
```
{}

```

**Comment by j6s** [comment]
That's very interesting. Between first implementing `mailcow-exporter` and now my instance was unblocked by another server and I can confirm, that the endpoint will return an array, if there are queued items but and empty JSON object if there is nothing in the queue now. This is a bug in the mailcow API, but I will see how I can also handle it in the exporter.

Thanks for the help debugging.

**Comment by j6s** [comment]
[01db707] Should resolve this issue by ignoring API response errors. This is not ideal, but until fixed on the mailcow side it should do the job.

**Comment by badsmoke** [comment]
I have to thank you for the quick feedback

I have built a rspamd_exporter for mailcows rspamd, so I am very interested in this exporter :-)



with the latest version, some data arrives now but not all, but hopefully this will change when mailcow reacts to your issue

**Comment by j6s** [comment]
Interesting - also: As part of https://github.com/mailcow/mailcow-dockerized/issues/1695 a new `/api/v1/get/logs/rspamd-stats` has been implemented. I haven't looked at it yet, but it could be promising

**Comment by badsmoke** [comment]
very cool thanks for the hint, I did not even notice
